ocaml/testsuite/tests/typing-layouts/literals.ml

```diff
@@ -7,150 +7,142 @@
-1 | let positive_one = +#1l
-                       ^^^^
-Error: Unboxed int literals aren't supported yet.
+positive_one: 1
 |}]
-
-let positive_one = + #1l
+let () = test_int32 "positive_one" (+ #1l)
 [%%expect{|
-Line 1, characters 19-24:
-1 | let positive_one = + #1l
-                       ^^^^^
-Error: Unboxed int literals aren't supported yet.
+positive_one: 1
 |}]
-
-let negative_one = -#1L
+let () = test_int64 "negative_one" (-#1L)
 [%%expect{|
-Line 1, characters 19-23:
-1 | let negative_one = -#1L
-                       ^^^^
-Error: Unboxed int literals aren't supported yet.
+negative_one: -1
 |}]
-
-let negative_one = - #1L
+let () = test_int64 "negative_one" (- #1L)
 [%%expect{|
-Line 1, characters 19-24:
-1 | let negative_one = - #1L
-                       ^^^^^
-Error: Unboxed int literals aren't supported yet.
+negative_one: -1
 |}]
-
-let two_fifty_five_in_hex = #0xFFn
+let () = test_nativeint "two_fifty_five_in_hex" (#0xFFn)
 [%%expect{|
-Line 1, characters 28-34:
-1 | let two_fifty_five_in_hex = #0xFFn
-                                ^^^^^^
-Error: Unboxed int literals aren't supported yet.
+two_fifty_five_in_hex: 255
 |}]
-
-let twenty_five_in_octal = #0o31l
+let () = test_int32 "twenty_five_in_octal" (#0o31l)
 [%%expect{|
-Line 1, characters 27-33:
-1 | let twenty_five_in_octal = #0o31l
-                               ^^^^^^
-Error: Unboxed int literals aren't supported yet.
+twenty_five_in_octal: 25
 |}]
-
-let forty_two_in_binary = #0b101010L
+let () = test_int64 "forty_two_in_binary" (#0b101010L)
 [%%expect{|
-Line 1, characters 26-36:
-1 | let forty_two_in_binary = #0b101010L
-                              ^^^^^^^^^^
-Error: Unboxed int literals aren't supported yet.
+forty_two_in_binary: 42
 |}]
 
-let () = test "one_twenty_seven_point_two_five_in_floating_hex" (#0x7f.4)
+let () = test_float "one_twenty_seven_point_two_five_in_floating_hex" (#0x7f.4)
 [%%expect{|
 one_twenty_seven_point_two_five_in_floating_hex: 127.250000
 |}]
 
-let () = test "five_point_three_seven_five_in_floating_hexponent" (#0xa.cp-1)
+let () = test_float "five_point_three_seven_five_in_floating_hexponent" (#0xa.cp-1)
 [%%expect{|
 five_point_three_seven_five_in_floating_hexponent: 5.375000
 |}]
 
-let () = test "unknown_floating_point_suffix" (#0.P)
+let () = test_float "unknown_floating_point_suffix" (#0.P)
 [%%expect{|
-Line 1, characters 46-52:
-1 | let () = test "unknown_floating_point_suffix" (#0.P)
-                                                  ^^^^^^
+Line 1, characters 52-58:
+1 | let () = test_float "unknown_floating_point_suffix" (#0.P)
+                                                        ^^^^^^
 Error: Unknown modifier 'P' for literal #0.P
 |}]
 
+let () = test_int32 "unknown_floating_point_suffix" (#0g)
+[%%expect{|
+Line 1, characters 52-57:
+1 | let () = test_int32 "unknown_floating_point_suffix" (#0g)
+                                                        ^^^^^
+Error: Unknown modifier 'g' for literal #0g
+|}]
```

Comment:
`unknown_int_suffix`? (I'd make a GitHub suggestion but I'm sure I'd get the column numbers wrong...)